src/main/java/org/codeacademy/siuntupristatymas/controller/CourierController.java

```diff
@@ -0,0 +1,71 @@
+package org.codeacademy.siuntupristatymas.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.codeacademy.siuntupristatymas.dto.CreateCourierRequest;
+import org.codeacademy.siuntupristatymas.dto.GetCourierParcelsResponse;
+import org.codeacademy.siuntupristatymas.dto.GetCourierResponse;
+import org.codeacademy.siuntupristatymas.entity.Courier;
+import org.codeacademy.siuntupristatymas.mapper.CourierMapper;
+import org.codeacademy.siuntupristatymas.mapper.ParcelMapper;
+import org.codeacademy.siuntupristatymas.service.CourierService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RequiredArgsConstructor
+@RequestMapping("/api/couriers")
+@RestController
+public class CourierController {
+
+    private final CourierService courierService;
+    private final CourierMapper courierMapper;
+    private final ParcelMapper parcelMapper;
+
+    @GetMapping
+    public ResponseEntity<List<GetCourierResponse>> getAllCouriers(@RequestParam(required = false) Long id) {
+        List<GetCourierResponse> couriers = courierMapper.courierListToDto(courierService.getAllCouriers());
+        if (couriers.isEmpty()) {
+            return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+        }
+        return ResponseEntity.ok(couriers);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<GetCourierResponse> getCourierById(@PathVariable Long id) {
+        Courier courierById = courierService.getCourierById(id);
+        GetCourierResponse response = courierMapper.courierToDto(courierById);
+        return ResponseEntity.ok(response);
+    }
+
+    @GetMapping("/parcels/{id}")
```

Comment:
```suggestion
    @GetMapping("/{id}/parcels")
```

Ne siuntą pagal ID imi, o kurjerio pagal ID siuntas